asciidoc/session-API.adoc

```diff
@@ -0,0 +1,682 @@
+
+An auto-commit transaction is a basic but limited form of transaction.
+Such a transaction consists of only one Cypher query and is not automatically replayed on failure.
+Therefore any error scenarios will need to be handled by the client application itself.
+
+Auto-commit transactions are intended to be used for simple use cases such as when learning Cypher or writing one-off scripts.
+It is not recommended to use auto-commit transactions in production environments.
+
+[NOTE]
+The only way to execute `PERIODIC COMMIT` Cypher queries is to auto-commit the transaction.
+Unlike other kinds of Cypher query, `PERIODIC COMMIT` queries do not participate in the causal chain.
+Please refer to the <<cypher-manual#query-using-periodic-commit-hint, Cypher Manual -> `PERIODIC COMMIT` query hint>>.
+
+[.tabbed-example]
+.Simple auto-commit transaction example
+====
+[.include-with-dotnet]
+======
+
+[source, csharp]
+----
+include::{dotnet-examples}/Examples.cs[tags=autocommit-transaction]
+----
+======
+
+// [.include-with-go]
+// ======
+//
+// [source, go]
+// ----
+// include::{go-examples}/examples_test.go[tags=autocommit-transaction]
+// ----
+// ======
+
+[.include-with-java]
+======
+
+[source, java]
+----
+include::{java-examples}/AutocommitTransactionExample.java[tags=autocommit-transaction]
+----
+======
+
+[.include-with-javascript]
+======
+
+[source, javascript]
+----
+include::{javascript-examples}/examples.test.js[tags=autocommit-transaction]
+----
+======
+
+// [.include-with-python]
+// ======
+//
+// [source, python]
+// ----
+// include::{python-examples}/autocommit_transaction_example.py[tags=autocommit-transaction]
+// ----
+// ======
+====
+
+[[driver-simple-result-consume]]
+=== Consuming results
+
+Query results are typically consumed as a stream of records.
+The drivers provide a language-idiomatic way to iterate through that stream.
+
+[.tabbed-example]
+.Consuming the stream
+====
+[.include-with-dotnet]
+======
+[source, csharp]
+----
+include::{dotnet-examples}/Examples.cs[tags=result-consume]
+----
+======
+
+// [.include-with-go]
+// ======
+// [source, go]
+// ----
+// include::{go-examples}/examples_test.go[tags=result-consume]
+// ----
+// ======
+
+[.include-with-java]
+======
+[source, java]
+----
+include::{java-examples}/ResultConsumeExample.java[tags=result-consume]
+----
+======
+
+[.include-with-javascript]
+======
+[source, javascript]
+----
+include::{javascript-examples}/examples.test.js[tags=result-consume]
+----
+======
+
+// [.include-with-python]
+// ======
+// [source, python]
+// ----
+// include::{python-examples}/result_consume_example.py[tags=result-consume]
+// ----
+// ======
+====
+
+
+[[driver-simple-result-retain]]
+=== Retaining results
+
+Within a session, only one result stream can be active at any one time.
+Therefore, if the result of one query is not fully consumed before another query is executed, the remainder of the first result will be automatically buffered within the Result object.
```

Comment:
Why is "result object" capitalized here?

Reply:
fixed